pkg/ucp/backend/service.go

```diff
@@ -19,83 +19,70 @@ package backend
 import (
 	"context"
 	"errors"
-	"fmt"
 	"net/http"
 	"net/url"
 
 	v1 "github.com/radius-project/radius/pkg/armrpc/api/v1"
 	ctrl "github.com/radius-project/radius/pkg/armrpc/asyncoperation/controller"
 	"github.com/radius-project/radius/pkg/armrpc/asyncoperation/worker"
-	"github.com/radius-project/radius/pkg/armrpc/hostoptions"
 	"github.com/radius-project/radius/pkg/sdk"
+	"github.com/radius-project/radius/pkg/ucp"
 	"github.com/radius-project/radius/pkg/ucp/api/v20231001preview"
 	"github.com/radius-project/radius/pkg/ucp/backend/controller/resourcegroups"
 	"github.com/radius-project/radius/pkg/ucp/backend/controller/resourceproviders"
 	"github.com/radius-project/radius/pkg/ucp/datamodel"
-	ucpoptions "github.com/radius-project/radius/pkg/ucp/hostoptions"
 	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
 )
 
-const (
-	UCPProviderName = "System.Resources"
-)
-
 // Service is a service to run AsyncReqeustProcessWorker.
 type Service struct {
 	worker.Service
-
-	config ucpoptions.UCPConfig
+	options *ucp.Options
 }
 
-// NewService creates new service instance to run AsyncRequestProcessWorker.
-func NewService(options hostoptions.HostOptions, config ucpoptions.UCPConfig) *Service {
+// NewService creates new backend service instance to run the async worker.
+func NewService(options *ucp.Options) *Service {
+	workerOptions := worker.Options{}
+	if options.Config.Worker.MaxOperationConcurrency != nil {
+		workerOptions.MaxOperationConcurrency = *options.Config.Worker.MaxOperationConcurrency
+	}
+	if options.Config.Worker.MaxOperationRetryCount != nil {
+		workerOptions.MaxOperationRetryCount = *options.Config.Worker.MaxOperationRetryCount
+	}
 	return &Service{
+		options: options,
 		Service: worker.Service{
-			ProviderName: UCPProviderName,
-			Options:      options,
+			OperationStatusManager: options.StatusManager,
+			Options:                workerOptions,
+			QueueProvider:          options.QueueProvider,
+			StorageProvider:        options.StorageProvider,
 		},
-		config: config,
 	}
 }
 
-// Name returns a string containing the UCPProviderName and the text "async worker".
```

Comment:
Probably AI generated comment :)

Reply:
I delete the AI-generated comments when I update code that has them. They are usually wrong.